### PR TITLE
Make some changes to the postgres database module

### PR DESCRIPTION
**Pull request by ajitdsa**
# Description

- clarify some items in the postgres.md docs
- change `data` field from text to jsonb for indexing purposes
- do some simplicity-based data manipulation on put and get functions
- add in a database-wide table_name config that is overrideable at the skill level

## Status
**READY**

## Type of change

- New feature (non-breaking change which adds functionality)
- Documentation (fix or adds documentation)

# How Has This Been Tested?

Not tested yet :/

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/databases/postgres.md [hunk 1]
```diff
@@ -13,13 +13,14 @@ databases:
   postgresql:
     host: "hostname" # (optional) default: "localhost"
     port: 5432 # (optional) default: 5432
-    user: opsdroid # (optional) default: opsdroid
+    user: "opsdroid" # (optional) default: "opsdroid"
     password: "Please change me"
-    database: opsdroid_db # (optional) default: opsdroid
+    database: "opsdroid_db" # (optional) default: "opsdroid"
+    table_name: "opsdroid_table" # (optional) default: "opsdroid_default"
```
Comment (dud1337) [inline docs/databases/postgres.md]: suggest `default_table_name` as a var

File: opsdroid/database/postgresql/__init__.py [hunk 2]
```diff
@@ -65,7 +68,9 @@ def __init__(self, config, opsdroid=None):
         self.user = self.config.get("user", "opsdroid")
         self.password = self.config.get("password")
         self.database = self.config.get("database", "opsdroid")
+        self.table_name = self.config.get("table_name", "opsdroid_default")
         self.host = self.config.get("host", "localhost")
+        self.port = self.config.get("port", 5432)
```
Comment (dud1337) [inline opsdroid/database/postgresql/__init__.py:73]: nice

File: opsdroid/database/postgresql/__init__.py [hunk 3]
```diff
@@ -27,7 +27,10 @@ async def wrapper(*args, **kwargs):
         async with connection.transaction():
             # Create table if it does not exist
             await connection.execute(
-                'CREATE TABLE IF NOT EXISTS "{}" ( key text PRIMARY KEY, data text)'.format(table_name)
+                'CREATE TABLE IF NOT EXISTS "{}" ( key text PRIMARY KEY, data JSONb)'.format(table_name)
+            )
+            await connection.execute(
+                'CREATE INDEX IF NOT EXISTS idxgin ON "{}" USING gin (data);'.format(table_name)
```
Comment (dud1337) [inline opsdroid/database/postgresql/__init__.py:33]: Why do we want JSONb and INDEX?

**Review by dud1337 — CHANGES_REQUESTED** [review]
Awesome. Added some comments

**Review by ajitdsa — COMMENTED** [review]
I did have a question for you @dud1337... On line 22 of opsdroid/database/postgresql/__init__.py ... ideally this would be the `default_table_name` set in the config. How would we get that into the decorator in some sensible way?

**Comment by dud1337** [comment]
Good question. I don't have an answer yet.

File: opsdroid/database/postgresql/__init__.py (hunk 3 above)
Comment (ajitdsa) [reply opsdroid/database/postgresql/__init__.py:33]: So we can't assume that this is the only application accessing the data. In fact, when I personally use this, I will be hooking up the database to a [Hasura GraphQL](https://hasura.io/) application. If you use an indexed jsonb field, you can write queries with `WHERE` clauses based on the data in the jsonb field. You can't do that with a text field. This makes it infinitely more useful 👍 (https://medium.com/hackernoon/how-to-query-jsonb-beginner-sheet-cheat-4da3aa5082a3)

**Pull request merged**